### PR TITLE
Added feature to use SAS uri for secret storage and supporting tests

### LINKED ISSUE #5190: Backport SAS support for secret manager repository to V1
The work done in V2 [here](https://github.com/Azure/azure-functions-host/pull/4723) needs to be backported to v1

### PR DESCRIPTION
**Pull request by gzuber**
Resolves #5190 

Backport of [this PR](https://github.com/Azure/azure-functions-host/pull/4723)

### CODE REVIEW

File: src/WebJobs.Script.WebHost/GlobalSuppressions.cs [hunk 1]
```diff
@@ -125,4 +125,9 @@
...
 [assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA1801:ReviewUnusedParameters", MessageId = "config", Scope = "member", Target = "Microsoft.Azure.WebJobs.Script.WebHost.Extensions.FunctionMetadataExtensions.#GetTestData(System.String,Microsoft.Azure.WebJobs.Script.ScriptHostConfiguration)")]
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors", Scope = "member", Target = "Microsoft.Azure.WebJobs.Script.WebHost.SecretManager.#.ctor(Microsoft.Azure.WebJobs.Script.WebHost.ISecretsRepository,Microsoft.Azure.WebJobs.Script.WebHost.IKeyValueConverterFactory,Microsoft.Azure.WebJobs.Host.TraceWriter,System.Boolean)")]
\ No newline at end of file
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Usage", "CA2214:DoNotCallOverridableMethodsInConstructors", Scope = "member", Target = "Microsoft.Azure.WebJobs.Script.WebHost.SecretManager.#.ctor(Microsoft.Azure.WebJobs.Script.WebHost.ISecretsRepository,Microsoft.Azure.WebJobs.Script.WebHost.IKeyValueConverterFactory,Microsoft.Azure.WebJobs.Host.TraceWriter,System.Boolean)")]
```
Comment (gzuber) [inline src/WebJobs.Script.WebHost/GlobalSuppressions.cs]: All the new supress messages in this PR are related to the stylecop thinking "Sas" is hungarian notation

File: src/WebJobs.Script.WebHost/Security/BlobStorageSecretsRepository.cs [hunk 2]
```diff
@@ -182,6 +188,7 @@ private void Dispose(bool disposing)
         public void Dispose()
         {
             Dispose(true);
+            GC.SuppressFinalize(this);
```
Comment (gzuber) [inline src/WebJobs.Script.WebHost/Security/BlobStorageSecretsRepository.cs]: This is unrelated, but for some reason the project wouldn't build without this change. This seems to match other Dispose patterns, though.

File: src/WebJobs.Script.WebHost/Security/DefaultSecretsRepositoryFactory.cs [hunk 3]
```diff
@@ -17,7 +17,13 @@ public ISecretsRepository Create(ScriptSettingsManager settingsManager, WebHostS
         {
             string secretStorageType = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebJobsSecretStorageType);
             string storageString = AmbientConnectionStringProvider.Instance.GetConnectionString(ConnectionStringNames.Storage);
-            if (secretStorageType != null && secretStorageType.Equals("Blob", StringComparison.OrdinalIgnoreCase) && storageString != null)
+            string secretStorageSas = settingsManager.GetSetting(EnvironmentSettingNames.AzureWebJobsSecretStorageSas);
```
Comment (gzuber) [inline src/WebJobs.Script.WebHost/Security/DefaultSecretsRepositoryFactory.cs]: I believe this change should preserve the current behavior if SecretStorageSas isn't provided.

File: src/WebJobs.Script.WebHost/Security/BlobStorageSecretsRepository.cs (hunk 2 above)
Comment (ahmelsayed) [reply src/WebJobs.Script.WebHost/Security/BlobStorageSecretsRepository.cs]: Dispose pattern in dotnet requires that  https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Comment (gzuber) [reply src/WebJobs.Script.WebHost/Security/BlobStorageSecretsRepository.cs]: oh that's a nice resource, thanks!

File: src/WebJobs.Script.WebHost/Security/BlobStorageSasSecretsRepository.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Autofac.Core;
```
Comment (fabiocav) [inline src/WebJobs.Script.WebHost/Security/BlobStorageSasSecretsRepository.cs]: nit: remove unused "usings" and reorder

**Pull request merged**